### PR TITLE
Add initial support for python 3.13

### LINKED ISSUE #6232: Windows on ARM missing distlib binaries in pyproject.toml
### Issue description

Unable to install packages into pipenv environment on Windows on ARM

### Expected result

Install requirements successfully

### Actual result

ValueError: Unable to find resource t64-arm.exe in package pipenv.patched.pip._vendor.distlib

### Steps to replicate

Install almost packages with pipenv on Windows on ARM (example: wheel, setuptools, semantic-version, setuptools-rust)

### Steps to fix the issue

Edit pipenv's pyproject.toml with the 't64-arm.exe' and 'w64-arm.exe' added for distlib:

"pipenv.patched.pip._vendor.distlib" = [
  "t32.exe",
  "t64.exe",
  "t64-arm.exe",
  "w32.exe",
  "w64.exe",
  "w64-arm.exe",
]

### PR DESCRIPTION
**Pull request by matteius**
* Initial support for python3.13
* Fixes #6232

### The checklist

* [X] Associated issue
* [X] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix.rst`, `.feature.rst`, `.behavior.rst`, `.doc.rst`. `.vendor.rst`. or `.trivial.rst` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -29,6 +28,7 @@ pyyaml = "==6.0.1"
 build = "*"
 twine = "*"
 semver = "*"
+pypiserver = {ref = "pipenv-313", git = "https://github.com/matteius/pypiserver.git"}
```
Comment (matteius) [inline Pipfile:31]: https://github.com/pypiserver/pypiserver/pull/602

File: tests/integration/test_cli.py [hunk 2]
```diff
@@ -130,11 +130,7 @@ def test_pipenv_graph_reverse(pipenv_instance_private_pypi):
                                          output,
                                          flags=re.MULTILINE | re.IGNORECASE)
                 assert openpyxl_dep is None, f'openpyxl should not appear at beginning of lines in {output}'
-
-                if sys.version_info[:2] == (3, 12):
-                    assert 'openpyxl==2.5.4 [requires: et_xmlfile]' in output
-                else:
-                    assert 'openpyxl==2.5.4 [requires: et-xmlfile]' in output
+                assert 'openpyxl==2.5.4 [requires: et_xmlfile]' in output
```
Comment (matteius) [inline tests/integration/test_cli.py:133]: I have no idea how this straightened itself out.   I assume something to do with importlib-metadata updating, but really confusing.

File: pipenv/utils/indexes.py [hunk 3]
```diff
@@ -90,9 +90,11 @@ def get_source_list(
...
     if pypi_mirror:
         sources = [
-            create_mirror_source(pypi_mirror, source["name"])
-            if is_pypi_url(source["url"])
-            else source
+            (
+                create_mirror_source(pypi_mirror, source["name"])
+                if is_pypi_url(source["url"])
+                else source
+            )
```
[the remark this answers is not in the document]
Comment (matteius) [reply pipenv/utils/indexes.py:97]: because I upgraded black and it has updated rules.

[the remark this answers is not in the document]
Comment (oz123) [reply pipenv/utils/indexes.py:97]: Weird.

**Pull request merged**